### ISSUE TITLE
Incorrect criteria for showing the full screen modal

### ISSUE THREAD

**Issue by jthrilly** [codaco/Network-Canvas#266]
The full screen modal and new node form currently display incorrectly. We want this to be for tablet devices with limited screen space, but _only_ where they do not have a hardware keyboard connected. For example, the iPad pro without the keyboard connected should display the full screen form (because it will use the software keyboard), but with the keyboard connected it should display the normal form.

AFAIK, this is not possible, but I want to at least try and investigate it.

Currently we use aspect ratio, but this is inadequate because the hardware keyboard takes up vertical height, not horizontal. 

Ideas:

- Can we use cordova somehow? Is there a plugin to detect if there is a HW keyboard?
- Should we fall back on having this be a global protocol option that is set by the researcher and can be toggled during an interview?
- Can we somehow detect the resize event when the software keyboard is shown?

**Comment by jthrilly** [comment]
I can no longer reproduce this, so closing for now.

**Comment by jthrilly** [comment]
This is a combination of issues: screen size, and physical keyboard status.

Examples of devices that should not have the full screen form mode but do:

- 15" Macbook Pro (large enough screen and a physical keyboard)
- 13" Macbook Pro (as above)
- iPad Pro with keyboard connected. (has physical keyboard)

Since there is no way (known way?) to detect a physical keyboard, I suggest we resolve this after #390 by adding a global setting for if the full screen modal should be used. We can do a "best guess" setting on first run based on what we know about the device.

**Comment by rebeccamadsen** [comment]
#390 is currently in the Alpha.4 milestone. Should this be also, or should #390 be in Alpha.3?

**Comment by jthrilly** [comment]
#390 is really independent of this, since creating a global app setting for this feature is probably a good idea regardless of how/if we can resolve the automatic setting.

Basically, the ideal resolution to this issue will give us a better 'default' for #390.

**Comment by rebeccamadsen** [comment]
Okay. I haven't been able to find a way to detect a physical keyboard in cordova. There were a few that might have been something we could leverage, but nothing that supported android and ios that I've found.

We ought to be able to detect if a soft keyboard is showing, but that can show/hide just by users clicking on the screen anywhere other than the input field or soft keyboard. I think it would be very strange to change the form on them just when they tap on the screen somewhere, especially because we wouldn't know where in the form wizard they should be plunked.

With that in mind, I wonder if we could find a better option just by considering the screen information we *can* reliably access. For example, these are the specs I found:

* 15" macbook pro (2880x1800) = 1.6 aspect ratio
* 13" macbook pro (2560x1600) = 1.6 aspect ratio
* ipad pro (1920x1080) = 1.7 aspect ratio
* ipad pro 12.5" (2732x2048) = 1.3 aspect ratio
* ipad pro 10.5" (2224x1668) = 1.3 aspect ratio

We currently have the cutoff for showing the full screen form mode `min-device-aspect-ratio` below 16/9, which looks at the device aspect ratio so that the keyboard show/hide doesn't change this number. It encompasses almost all of these (excepting the ipad pro with the 1920x1080 resolution, which is an edge case and might round differently?). 

Maybe we could drop the aspect ratio to 1.6 or looking at some combo of aspect ratio and device height?
